src/components/Navigation/AuthenticatedNavBar.js

```diff
@@ -0,0 +1,103 @@
+import React, {useCallback, useState} from "react";
+import PropTypes from "prop-types";
+
+import {logout} from "actions/authActions";
+import {useDispatch, useSelector} from "react-redux";
+import {useHistory} from "react-router-dom";
+
+import Hidden from "@material-ui/core/Hidden";
+import IconButton from "@material-ui/core/IconButton";
+import LogoutIcon from "@material-ui/icons/ExitToAppOutlined";
+import MenuIcon from "@material-ui/icons/Menu";
+import NavLinkNoDup from "../Routes/NavLinkNoDup";
+import Search from "components/FeatureViews/Search/Search";
+import Toolbar from "@material-ui/core/Toolbar";
+import AppBar from "@material-ui/core/AppBar";
+import Typography from "@material-ui/core/Typography";
+// Local Component Imports
+import Avatar from "@material-ui/core/Avatar";
+import {stringToColor} from "../FeatureViews/Accounts/accountUtils";
+import Tooltip from "@material-ui/core/Tooltip";
+import {fullName} from "../../utils";
+import {closeRegistrationCart} from "../OmouComponents/RegistrationUtils";
+
+const AuthenticatedNavBar = ({toggleDrawer}) => {
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const authUser = useSelector(({auth}) => auth);
+
+    const [mobileOpen, setMobileOpen] = useState(false);
+    const [isMobileSearching, setMobileSearching] = useState(false);
+
+    const handleDrawerToggle = useCallback(() => {
+        toggleDrawer(!mobileOpen);
+        setMobileOpen((open) => !open);
+    }, [mobileOpen, toggleDrawer]);
+
+    const handleLogout = useCallback(() => {
+        closeRegistrationCart();
+        dispatch(logout());
+        history.push("/login");
+    }, [dispatch, history]);
+
+    const handleMobileSearch = useCallback((searchQuery) => {
+        setMobileSearching(searchQuery);
+    }, []);
+
+    const name = fullName(authUser.user);
+
+    return (
+        <AppBar className="OmouBar" position="sticky">
+            <Toolbar>
+                {!isMobileSearching && (
+                    <>
+                        <Hidden lgUp>
+                            <IconButton aria-label="Open Drawer" style={{color: "white"}}
```

Comment:
Is this a button we would want to migrate to ResponsiveButton as well?

Reply:
This is an icon so we'll want to use IconButton specifically - good catch though :)